sbi/neural_nets/estimators/categorical_net.py

```diff
@@ -159,6 +159,11 @@ def sample(
 
         return samples.reshape(*sample_shape, batch_dim, self.num_variables)
 
+    def to(self, device):
+        """Override `to()` to move self.mask along with the model."""
+        self.mask = self.mask.to(device)
```

Comment:
Usually, one registers non-differentiable/static but necessary parts as "Buffers" in PyTorch modules. Buffers will be automatically moved onto the device i.e. in the init do `self.register_buffer("mask", mask)` instead.

Reply:
oh yes, very good point! will change it to that.